### DISCUSSION TITLE
Inav and Walksnail WS-M181 GPS

### DISCUSSION THREAD

**Discussion by Normoa** [iNavFlight/inav#10481] [Support]
Hi,

I'm trying the new M10 GPS module, a WS-M181 GPS.
It's tiny, and seems quite good.
The question I have is about the magnetometer. 
Does anyone know the alignment for the module, and what the alignment config should be set in Inav?
Can't find the info anywhere, and haven't had a reply from Caddx support.

Cheers.

**Comment by sensei-hacker** [comment]
If you don't get an answer from someone else who has that particular module, the "emperical method" described on the wiki page always works.

Most modules have the default alignment when the plug in facing forward. Which means you may need to rotate the setting 180° in yaw if the  GPS/mag is mounted away from your antennas and power and cable if it's mounted up front above the camera.

**Comment by Normoa** [comment]
Hiya,

Thanks for taking the time to respond.
My normal module - BN-880, is aligned when the plug is at the back of the module, and then using CW270-flip as the setting.

Cheers.

**Reply by Normoa** [reply]
I think the correct process is -

1) Calibrate the mag.
2) Point the nose of the quad at North.
3) If the setup page info doesn't show it pointing north, then adjust the mag alignment info (may need a couple of goes) until it says north (0 degrees - or darned close to it). 
4) Save the setup, and then double check it.

Does that seem about right?

**Reply by sensei-hacker** [reply]
> I think the correct process is -
> 
> 1. Calibrate the mag.
> 2. Point the nose of the quad at North.
> 3. If the setup page info doesn't show it pointing north, then adjust the mag alignment info (may need a couple of goes) until it says north (0 degrees - or darned close to it).
> 4. Save the setup, and then double check it.
> 
> Does that seem about right?



That's half of it.  It can be pointed north correctly, but configured upside-down. This also needs to match up with what the gyro thinks is clockwise vs counter clockwise rotation.
See
https://github.com/iNavFlight/inav/wiki/GPS-and-Compass-setup#emperical-method

**Reply by Normoa** [reply]
Most excellent reply, sir!

Thank you for your effort (again).

Yep, I always ensure the ceramic antenna is uppermost.

Good point about checking that the rotation is correct.

And the Wiki even documents the Walksnail GPS (which I've been unable to find anywhere - not even from the manufacturer).

I've bookmarked the Wiki (for some reason I always have trouble trying to find it, when I go looking through github)

Again, many thanks - I redo the setup, checking with the Wiki.
